src/Tools/dotnet-monitor/HostBuilderHelper.cs

```diff
@@ -18,76 +18,39 @@ internal static class HostBuilderHelper
-            = "DotnetMonitorTestSettings__UserConfigDirectoryOverride";
-
-        // Location where shared dotnet-monitor configuration is stored.
-        // Windows: "%ProgramData%\dotnet-monitor
-        // Other: /etc/dotnet-monitor
-        private static readonly string SharedConfigDirectoryPath =
-            GetEnvironmentOverrideOrValue(
-                SharedConfigDirectoryOverrideEnvironmentVariable,
-                RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
-                    Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData), ProductFolderName) :
-                    Path.Combine("/etc", ProductFolderName));
-
-        private static readonly string SharedSettingsPath = Path.Combine(SharedConfigDirectoryPath, SettingsFileName);
-
-        // Location where user's dotnet-monitor configuration is stored.
-        // Windows: "%USERPROFILE%\.dotnet-monitor"
-        // Other: "%XDG_CONFIG_HOME%/dotnet-monitor" OR "%HOME%/.config/dotnet-monitor"
-        private static readonly string UserConfigDirectoryPath =
-            GetEnvironmentOverrideOrValue(
-                UserConfigDirectoryOverrideEnvironmentVariable,
-                RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
-                    Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), "." + ProductFolderName) :
-                    Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), ProductFolderName));
-
-        private static readonly string UserSettingsPath = Path.Combine(UserConfigDirectoryPath, SettingsFileName);
-
-        public static IHostBuilder CreateHostBuilder(string[] urls, string[] metricUrls, bool metrics, string diagnosticPort, bool noAuth, bool tempApiKey)
-        {
-            return CreateHostBuilder(urls, metricUrls, metrics, diagnosticPort, CreateAuthConfiguration(noAuth, tempApiKey));
-        }
 
-        public static IHostBuilder CreateHostBuilder(string[] urls, string[] metricUrls, bool metrics, string diagnosticPort, AuthConfiguration authenticationOptions)
+        public static IHostBuilder CreateHostBuilder(HostBuilderSettings settings)
         {
             return Host.CreateDefaultBuilder()
-                .UseContentRoot(AppContext.BaseDirectory) // Use the application root instead of the current directory
+                .UseContentRoot(settings.ContentRootDirectory)
                 .ConfigureHostConfiguration((IConfigurationBuilder builder) =>
                 {
                     //Note these are in precedence order.
-                    ConfigureEndpointInfoSource(builder, diagnosticPort);
-                    ConfigureMetricsEndpoint(builder, metrics, metricUrls);
+                    ConfigureEndpointInfoSource(builder, settings.DiagnosticPort);
+                    ConfigureMetricsEndpoint(builder, settings.EnableMetrics, settings.MetricsUrls ?? Array.Empty<string>());
```

Comment:
Is there a reason to do the `?? Array.Empty<string>()` here as opposed to in `CreateMonitor` (this also applies to urls)? I don't have any real issue with this - I was mostly just curious why we didn't deal with the issue at the source on creation.

Reply:
It simplifies the `HostBuilderSettings` class so that it doesn't have to validate that the property is not null. Couple of options are:
1. Leave as written.
2. Add validation to `HostBuilderSettings` to make sure it is not null.
3. Change `Urls` and `MetricsUrls` to be readonly properties with an empty `List<string>` as its default value. Then call `ToArray` or update downstream to handle `List<string>` instead.

I personally like options 1 or 3, neither requires extra code to validate something that is very unlikely to happen in real usage.

Reply:
Of the two, option 1 is better at this time because we get the urls as `string[]` and can pass them directly through to the settings. Option 3 would require iterating/copying the array (which isn't going to be too expensive since we're only dealing with probably less than three items 99% of the time).